### PR TITLE
Disable SAC factor setting for CCR dives

### PR DESCRIPTION
**Pull request by atdotde**
The SAC factor is only used for minimal gas calculations which
don't make sense in the CCR context.

Additionally, make bailout stop for at least minimum switch
time or problem solving time.

Signed-off-by: Robert C. Helling <helling@atdotde.de>

This is a small addition that had been sitting on my hard drive for a while but which could go into 4.9.7

### Describe the pull request:

- [ ] Bug fix
- [x] Functional change
- [ ] New feature
- [ ] Code cleanup
- [ ] Build system change
- [ ] Documentation change
- [ ] Language translation

### CODE REVIEW

File: desktop-widgets/diveplanner.cpp [hunk 1]
```diff
@@ -606,6 +606,8 @@ void PlannerSettingsWidget::setBackgasBreaks(bool dobreaks)
 void PlannerSettingsWidget::setBailoutVisibility(int mode)
 {
 		ui.bailout->setDisabled(!(mode == CCR || mode == PSCR));
+		ui.sacFactor->setDisabled(mode == CCR);
+		ui.sacfactor->setDisabled(mode == CCR);
```
Comment (bstoeger) [inline desktop-widgets/diveplanner.cpp]: Because once is not enough? ;)

File: core/planner.c [hunk 2]
```diff
@@ -825,11 +825,12 @@ bool plan(struct deco_state *ds, struct diveplan *diveplan, struct dive *dive, i
 	if ((divemode == CCR || divemode == PSCR) && prefs.dobailout) {
 		divemode = OC;
 		po2 = 0;
+		int bailoutsegment = prefs.min_switch_duration < prefs.problemsolvingtime * 60 ? prefs.problemsolvingtime * 60 : prefs.min_switch_duration;
```
Comment (bstoeger) [inline core/planner.c]: What about
```
		int bailoutsegment = MAX(prefs.problemsolvingtime * 60, prefs.min_switch_duration);
```
to make this understandable without mental gymnastics?

[Completely unrelated: the things you have to do to define a proper MAX macro ... jeez. And that's probably not even standard C.]

File: desktop-widgets/diveplanner.cpp (hunk 1 above)
Comment (dirkhh) [reply desktop-widgets/diveplanner.cpp]: That way the silly thing is REALLY disabled 🤣

**Comment by dirkhh** [comment]
Happy to merge this before 4.9.7 - maybe after you look at @bstoeger 's comments :-)

File: desktop-widgets/diveplanner.cpp (hunk 1 above)
Comment (atdotde) [reply desktop-widgets/diveplanner.cpp]: I think you could be more trusting with respect to the optimisations modern compilers can do.

**Review by dirkhh — APPROVED** [review]

**Pull request merged**